pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -578,3 +582,225 @@ func TestFlowAggregator_closeUpdateChBeforeFlowExportLoopReturns(t *testing.T) {
 	close(stopCh2)
 	wg2.Wait()
 }
+
+func TestFlowAggregator_podInfoIndexFunc(t *testing.T) {
+	node := &v1.Node{}
+	pendingPod := &v1.Pod{
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.2",
+				},
+			},
+		},
+	}
+	succeededPod := &v1.Pod{
+		Status: v1.PodStatus{
+			Phase: v1.PodSucceeded,
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.3",
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name        string
+		obj         interface{}
+		want        []string
+		expectedErr string
+	}{
+		{
+			name:        "object is not pod",
+			obj:         node,
+			expectedErr: "obj is not pod: ",
+		},
+		{
+			name: "pod status in pending phase",
+			obj:  pendingPod,
+			want: []string{"192.168.1.2"},
+		},
+		{
+			name: "pod status in succeeded phase",
+			obj:  succeededPod,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := podInfoIndexFunc(tt.obj)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestFlowAggregator_fetchPodLabels(t *testing.T) {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "testPod",
+			Labels: map[string]string{
+				"test": "ut",
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.2",
+				},
+			},
+		},
+	}
+
+	client := fake.NewSimpleClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	informerFactory.Core().V1().Pods().Informer().AddIndexers(cache.Indexers{podInfoIndex: podInfoIndexFunc})
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	informerFactory.Start(stopCh)
+
+	tests := []struct {
+		name       string
+		podAddress string
+		addPod     bool
+		want       string
+	}{
+		{
+			name:       "no pod objects",
+			podAddress: "192.168.1.2",
+		},
+		{
+			name:       "pod with label",
+			podAddress: "192.168.1.2",
+			addPod:     true,
+			want:       "{\"test\":\"ut\"}",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			if tt.addPod {
+				informerFactory.Core().V1().Pods().Informer().GetIndexer().Add(pod)
+			}
+
+			fa := &flowAggregator{
+				k8sClient:        client,
+				includePodLabels: true,
+				podInformer:      informerFactory.Core().V1().Pods(),
+			}
+			got := fa.fetchPodLabels(tt.podAddress)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestFlowAggregator_GetRecordMetrics(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```

Comment:
this is no longer needer with recent gomock versions

Reply:
removed.